subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt

```diff
@@ -86,21 +88,44 @@ class SubscriptionFeedbackViewModel @Inject constructor(
     fun onReportTypeSelected(reportType: SubscriptionFeedbackReportType) {
         viewModelScope.launch {
             val previousFragmentState = viewState.value.currentFragmentState
-            val newMetadata = viewState.value.feedbackMetadata.copy(
+            var newMetadata = viewState.value.feedbackMetadata.copy(
                 reportType = reportType,
             )
```

Comment:
Would perhaps be easier to read, and more functional paradigm to keep it a `val` and fully construct here:

```suggestion
            val newMetadata = viewState.value.feedbackMetadata.let {
                it.copy(
                    reportType = reportType,
                    category = when (it.source) {
                        SUBSCRIPTION_SETTINGS -> SUBS_AND_PAYMENTS
                        VPN_MANAGEMENT, VPN_EXCLUDED_APPS -> VPN
                        else -> it.category
                    }
                )
            }
```

Then just use this `newMetadata` as normal down below, and we won't have to let it be reassigned as a side-effect of creating `nextState`.